### PR TITLE
Improve base repo (make it easier for participants)

### PR DESCRIPTION
**Pull request by joeytwiddle**
- Editor settings: Auto save, auto format, reduce size of terminal font

- Terminal improvements: auto restart server, auto rerun tests (thanks to nodemon)

- Use an `agent` in the user test instead of a `request`, so that cookies are preserved when redirecting.

- Move the `/users/welcome` route further down in the file, so it matches the flow of browser requests

- Use server-side session storage (which is tied to an auto-generated cookie) instead of client-side cookies for storage (because cookies can only be trusted if they contain a secret)

### CODE REVIEW

File: .theia/settings.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  "editor.tabSize": 2,
+  "editor.formatOnSave": true,
+  // But we disable formatting for ejs (html) because it prevents us from indenting <% if %>s
+  "[html]": {
+    "editor.formatOnSave": false
+  },
+  // Make editor and terminal font sizes match
+  "editor.fontSize": 12,
+  "terminal.integrated.fontSize": 12
+}
```
Comment (joeytwiddle) [inline .theia/settings.json:19]: Comments are fine in this file. Never mind what GitHub or Crockford think.

File: routes/index.js [hunk 2]
```diff
@@ -19,6 +19,10 @@ router.get('/', function (req, res, next) {
     }
   }
 
+  if (req.query.errorMsg) {
+    responseData.errorMsg = req.query.errorMsg
+  }
```
Comment (joeytwiddle) [inline routes/index.js:24]: Passing around error "tokens" and then turning them into human error messages might be good practice, but a bit cumbersome for this workshop.

File: .theia/settings.json (hunk 1 above)
Comment (miccheng) [reply .theia/settings.json:19]: LOL

File: README.md [hunk 3]
```diff
@@ -59,7 +59,7 @@ Open this folder in your terminal app. And run these commands in the terminal. T
 ## Architecture
 
 - **[Express](https://expressjs.com/)** - Application framework in NodeJS.
-- **[Pug](https://pugjs.org)** - Templating engine for displaying the HTML pages.
+- **[EJS](https://ejs.co/)** - Templating engine for displaying the HTML pages.
```
Comment (miccheng) [inline README.md:62]: Forgot to update this. Changed to EJS as reads like HTML. Less context switching.

File: routes/index.js (hunk 2 above)
Comment (miccheng) [reply routes/index.js:24]: OK. Was thinking of letting facilitators do that as a refactor. But yeah, this is fine.

**Review by miccheng — APPROVED** [review]
LGTM

**Pull request merged**